src/primitives/transaction.cpp

```diff
@@ -120,6 +121,7 @@ uint256 CTxInWitness::GetHash() const
     leaves.push_back(SerializeHash(vchIssuanceAmountRangeproof, SER_GETHASH, 0));
     leaves.push_back(SerializeHash(vchInflationKeysRangeproof, SER_GETHASH, 0));
     leaves.push_back(SerializeHash(scriptWitness.stack, SER_GETHASH, 0));
+    leaves.push_back(SerializeHash(pegin_witness.stack, SER_GETHASH, 0));
```

Comment:
Why can't we use the scriptWitness as the pegin_witness?

Reply:
what's the advantage of doing so? Seems to be abuse, since they serve two separate roles:
`scriptWitness`: utxo spending authorization
`pegin_witness`: peg-in authorization

Reply:
Avoiding serializing two witness arrays which are never used simultaneously.

Reply:
But they are. During a peg-in both are used. pegin_witness shows what the "scriptPubKey" is, then scriptWitness satisfies it.

Reply:
Oh, ok, I hadn't gotten that far yet. Never mind then :)